src/emscripten-optimizer/optimizer-shared.cpp

```diff
@@ -67,7 +56,7 @@ AsmType detectType(Ref node, AsmData *asmData, bool inVarDef, IString minifiedFr
   switch (node[0]->getCString()[0]) {
     case 'n': {
       if (node[0] == NUM) {
-        if (!isInteger(node[1]->getNumber())) return ASM_DOUBLE;
+        if (!wasm::isInteger(node[1]->getNumber())) return ASM_DOUBLE;
```

Comment:
Ah gotcha, I didn't realize that.

I am, but you have to make sure that every single contributor to emscripten_optimizer is OK licensing it to Apache 2.0 as well! Is that the case?

Having it in its own repo sounds best IMO, but then it indeed requires duplicating safe integer support or factoring that one into another repo as well. I think it would be best to export both in repos, and generate `.a` files from them.